packages/mui-material/src/Checkbox/Checkbox.js

```diff
@@ -55,7 +55,7 @@ const CheckboxRoot = styled(SwitchBase, {
         ? `rgba(${
             ownerState.color === 'default'
               ? theme.vars.palette.action.activeChannel
-              : theme.vars.palette.primary.mainChannel
+              : theme.vars.palette[ownerState.color].mainChannel
```

Comment:
```suggestion
              : theme.vars.palette[ownerState.color]?.mainChannel
```
I would add optional chaining here since the color value can be a custom value, otherwise, it will throw an error.

Reply:
But if the `color` value doesn't exist in theme, it will have invalid css. Is that acceptable?

Reply:
Just to see if I'm understanding it correctly — if folks pass another color that's not on the theme palette structure like `brand`, that's when the scenario is invalid, is that it?

Reply:
Yeah. This is usually tokens like `primary`, `secondary`, `success`, `warning` etc.

Reply:
We can check `theme.vars.palette[ownerState.color]` if it is not undefined, then use it. Otherwise, just skip it.

Reply:
Looking at the non css vars implementation, it does not validate the presence of `ownerState.color` in the theme palette. I think the current change should be fine, even without the `?` conditional check. If it is not defined, it'll make sure to throw early and let the dev know.

Reply:
@siriwatknp Merging as discussed.